### PR TITLE
[1.18] stats: fix stats when systemd cgroups are used

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
fixes cri stats for systemd cgroups

passing the slice directly to libpod.cgroups.Load() caused it to not be found. This is because it expected a fully qualified path, rather than a slice name.
write a function to translate slice name to path, and pass that down if we're using systemd cgroups
#### Does this PR introduce a user-facing change?

```release-note
Fixed a bug where systemd cgroups were incompatible with CRI stats endpoint
```

### CODE REVIEW

**Comment by haircommander** [comment]
```

not ok 35 ctr list filtering
# (in test file ./ctr.bats, line 647)
#   `[[ "$output" == "$ctr1_id" ]]' failed
# 0
```
I have never seen [that](https://storage.googleapis.com/origin-federated-results/pr-logs/pull/cri-o_cri-o/3398/test_pull_request_crio_integration_rhel/10576/artifacts/testout_userns.txt) before...

/retest

**Comment by haircommander** [comment]
we see the segfault here, but the fix is already upstream, and e2e_crun_cgroupv2 isn't blocking

@saschagrunert @umohnani8 @rhatdan @mrunalp PTAL

**Comment by rhatdan** [comment]
/approve
LGTM

File: internal/oci/oci_linux.go [hunk 1]
```diff
@@ -86,6 +87,16 @@ func newPipe() (parent, child *os.File, err error) {
 }
 
 func (r *runtimeOCI) containerStats(ctr *Container, cgroup string) (*ContainerStats, error) {
+	// this correction has to be made because the libpod cgroups package can't find a
+	// systemd cgroup that isn't converted to a fully qualified cgroup path
+	if r.config.CgroupManager == "systemd" {
```
Comment (saschagrunert) [inline internal/oci/oci_linux.go]:
```suggestion
	if r.config.CgroupManager == SystemdCgroupsManager {
```

File: internal/oci/oci_linux.go [hunk 2]
```diff
@@ -86,6 +87,16 @@ func newPipe() (parent, child *os.File, err error) {
 }
 
 func (r *runtimeOCI) containerStats(ctr *Container, cgroup string) (*ContainerStats, error) {
```
Comment (saschagrunert) [inline internal/oci/oci_linux.go]:
```suggestion
func (r *runtimeOCI) containerStats(ctr *Container, cgroup string) (stats *ContainerStats, err error) {
```

File: internal/oci/oci_linux.go [hunk 3]
```diff
@@ -86,6 +87,16 @@ func newPipe() (parent, child *os.File, err error) {
...
+	// this correction has to be made because the libpod cgroups package can't find a
+	// systemd cgroup that isn't converted to a fully qualified cgroup path
+	if r.config.CgroupManager == "systemd" {
+		var err error
```
Comment (saschagrunert) [inline internal/oci/oci_linux.go]:
```suggestion
```

**Pull request merged**